pallets/circuit/src/lib.rs

```diff
@@ -368,17 +369,32 @@ pub mod pallet {
                 None => CircuitStatus::Requested,
             };
 
-            let mut local_xtx_ctx: LocalXtxCtx<T> = Self::setup(
+            let local_xtx_ctx: LocalXtxCtx<T> = Self::setup(
                 fresh_or_revoked_exec.clone(),
                 &requester,
                 Zero::zero(),
                 maybe_xtx_id,
             )?;
-            Self::apply(&mut local_xtx_ctx, None, None)?;
+
+            // There should be no apply step since no change could have happen during the state access
+
+            let hardened_side_effects = local_xtx_ctx
+                .full_side_effects
+                .clone()
+                .iter()
+                .map(|step| {
+                    step.iter()
+                        .map(|fsx|
+                            HardenedSideEffect::convert(
+                                fsx.clone().harden().expect("Harden() shouldn't fail at this point after validation has been done before.")))
```

Comment:
We panic here, you can do this in multiple ways without being unsafe:
- collect results, partition and then return the errors
- use normal non-declarative flow and use error coalescence (?)

Reply:
yep yep will handle